src/eval_context.rs

```diff
@@ -196,7 +194,7 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
 
             Struct(_)    => unimplemented!(),
             Tuple(_)     => unimplemented!(),
-            Function(_)  => unimplemented!(),
+            Function(_, _)  => unimplemented!(),
```

Comment:
it's unreachable ^^ At least in the run-pass tests that we run. We only use this function in `mir::Literal::Value`, and I think that can only contain primitives